contracts/Aavegotchi/facets/ERC721BuyOrderFacet.sol

```diff
@@ -0,0 +1,190 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.1;
+
+import {LibAavegotchi} from "../libraries/LibAavegotchi.sol";
+import {LibBuyOrder} from "../libraries/LibBuyOrder.sol";
+import {LibERC721Marketplace} from "../libraries/LibERC721Marketplace.sol";
+import {LibERC20} from "../../shared/libraries/LibERC20.sol";
+import {IERC20} from "../../shared/interfaces/IERC20.sol";
+import {IERC721} from "../../shared/interfaces/IERC721.sol";
+import {LibMeta} from "../../shared/libraries/LibMeta.sol";
+import {Modifiers, ERC721BuyOrder} from "../libraries/LibAppStorage.sol";
+
+contract ERC721BuyOrderFacet is Modifiers {
+    event ERC721BuyOrderAdd(
+        uint256 indexed buyOrderId,
+        address indexed buyer,
+        address erc721TokenAddress,
+        uint256 erc721TokenId,
+        uint256 indexed category,
+        uint256 priceInWei,
+        uint256 time
+    );
+
+    event ERC721BuyOrderExecuted(
+        uint256 indexed buyOrderId,
+        address indexed buyer,
+        address seller,
+        address erc721TokenAddress,
+        uint256 erc721TokenId,
+        uint256 priceInWei,
+        uint256 time
+    );
+
+    function getERC721BuyOrder(uint256 _buyOrderId) external view returns (ERC721BuyOrder memory buyOrder_) {
+        buyOrder_ = s.erc721BuyOrders[_buyOrderId];
+        require(buyOrder_.timeCreated != 0, "ERC721BuyOrder: ERC721 buyOrder does not exist");
+    }
+
+    function getERC721BuyOrderIdsByTokenId(address _erc721TokenAddress, uint256 _erc721TokenId)
+        external
+        view
+        returns (uint256[] memory buyOrderIds_)
+    {
+        buyOrderIds_ = s.erc721TokenToBuyOrderIds[_erc721TokenAddress][_erc721TokenId];
+    }
+
+    function getERC721BuyOrdersByTokenId(address _erc721TokenAddress, uint256 _erc721TokenId)
+        external
+        view
+        returns (ERC721BuyOrder[] memory buyOrders_)
+    {
+        uint256[] memory buyOrderIds = s.erc721TokenToBuyOrderIds[_erc721TokenAddress][_erc721TokenId];
+        uint256 length = buyOrderIds.length;
+        buyOrders_ = new ERC721BuyOrder[](length);
+        for (uint256 i; i < length; i++) {
+            buyOrders_[i] = s.erc721BuyOrders[buyOrderIds[i]];
+        }
+    }
+
+    function placeERC721BuyOrder(
+        address _erc721TokenAddress,
+        uint256 _erc721TokenId,
+        uint256 _priceInWei,
+        bool[] calldata _validationOptions // 0: spirit force, 1: GHST, 2: skill points
```

Comment:
What if the user only wants to verify skill points, but not spirit force or GHST? And how about BRS / wearables?

Reply:
It's possible to verify one param only.
`_validationOptions` is an array of `bool`. You can check `LibBuyOrder.generateValidationHash` function details.
And for BRS, and wearables, we can add more validation logic to `LibBuyOrder.generateValidationHash` in the future.
When this is implemented, we discussed about validation of above 3 params only. You can check chat history on Discord.

Reply:
WDYM it's only possible to verify one param? What if they put `true` for all three of those params?

And I think we must have forgotten BRS, as that's probably the most important param to check. If BRS changes it's probably because the wearables changed, which means the buy order should probably be nullified. 

I think we should add BRS as a param too

Reply:
This function validate parameters who has `true` value only.
Let's add BRS value if you want.
Please confirm parameters to include in validation now.

Reply:
One issue I see with using Spirit Force as a parameter is that the am tokens increase in balance over balance, which means that param will fail immediately. So I think it's best if we remove Spirit Force. 

BRS is probably the most important parameter to validate, as it's related to wearables. So let's replace Spirit Force with BRS. I can't think of any other parameters that would need validating at the moment.

Reply:
OK.

Reply:
Done.